relengapi/blueprints/slaveloan/tasks.py

```diff
@@ -258,12 +258,43 @@ def waitfor_action(self, action_id, loanid):
         self.retry(exc=exc)
 
 
+@task(bind=True, max_retries=None)
+@add_to_history(
+    before="Calling slaveapi's disable method to disable from buildbot",
+    after="Disable request sent")
+def start_disable_slave(self, machine, loanid):
+    try:
+        url = furl(current_app.config.get("SLAVEAPI_URL", None))
+        url.path.add(machine).add("actions").add("shutdown_buildslave")
+        ret = retry(requests.post, args=(str(url),), ).json()
+        return (ret["requestid"], machine)
+    except Exception as exc:
+        logger.exception(exc)
+        self.retry(exc=exc)
+
+
+@task(bind=True, max_retries=None)
+@add_to_history(
+    after="Noticed that machine was disabled (or waiting timed out)")
+def waitfor_disable_slave(self, data, loanid):
+    requestid, machine = data
+    try:
+        url = furl(current_app.config.get("SLAVEAPI_URL", None))
+        url = furl(current_app.config.get("SLAVEAPI_URL", None))
+        url.path.add(machine).add("actions").add("shutdown_buildslave")
```

Comment:
Right, I was just confused (legitimately, I think) that a POST to
slaveapi/<machine>/actions/shutdown_buildslave does is a mechanism for
polling that a previously initiated operation is complete.  "POST",
"action" and "shutdown" all suggest that the request would, you know, shut
something down.  Principle of least surprise.